components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/JsGraphBuilder.java

```diff
@@ -588,6 +625,28 @@ private static void addEventListeners(DynamicDecisionNode decisionNode,
                 } else {
                     log.error("Event handler : " + key + " is not a function : " + value);
                 }
+            } else if (value instanceof SerializableJsFunction) {
+                decisionNode.addFunction(key, (SerializableJsFunction) value);
+            }
+        });
+    }
+
+    private static void addHandlers(ShowPromptNode showPromptNode, Map<String, Object> handlersMap) {
+
+        if (handlersMap == null) {
+            return;
+        }
+        handlersMap.forEach((key, value) -> {
+            if (value instanceof ScriptObjectMirror) {
```

Comment:
better to check ifFunction also

Reply:
This is checked from `SerializableJsFunction.toSerializableForm()`